hashserver_table.cpp

```diff
@@ -0,0 +1,29 @@
+#include "hashserver_table.hpp"
+#include <iostream>
+
+void hashserver_table::put(std::string key, std::string value)
+{   
+    std::lock_guard<std::mutex> lck(lock); 
+    key_value_table[key] = value;
+}
+
+std::string hashserver_table::get(std::string key)
+{   
+    std::string value_found;
+    auto search_key = key_value_table.find(key);
+    if (search_key == key_value_table.end())
+        return " ";
+    else {
+        value_found = search_key->second;
+        std::cout << "value: " << value_found << std::endl;
+        return value_found; 
+    }
+}
+
+int main(int argc, char* argv[]) {
```

Comment:
I think we want this either to be a loop, or store the key/value pair onto disk so that it can be reused across multiple invocations

Reply:
This is just the test code to verify the correctness of the hashtable. When we try to integrate with the Verena server or other clients who would like to write/read to hash_server, they just need to call get/put. This call can be over n/w or be part of the same application. We do not need to worry if this hashtable resides in memory or disk. for simplicity, we can assume the hashtable lies in memory.

Reply:
we need this code running independently (verena server will be a different application), so this needs to be its own server

in other words, we need to be able to execute this code multiple times, so the key/values need to be persisted somehow (in-memory is fine, but then this can't terminate until some other process kills it)

Reply:
As long as the application or server is running, the value will persist. Since we do not currently know how to communicate within two ser

Reply:
the hash server can't be the same application as the verena server; it could possibly be the same application as the client (for now), but if that's the case, we need to integrate this with the client code so they can run in the same execution loop

otherwise, this will need to handle its own application loop